sdk/servicebus/service-bus/src/sender.ts

```diff
@@ -307,18 +321,12 @@ export class SenderImpl implements Sender {
   async close(): Promise<void> {
     try {
       this._isClosed = true;
-      if (
-        this._context.namespace.connection &&
-        this._context.namespace.connection.isOpen() &&
-        this._context.sender
-      ) {
-        await this._context.sender.close();
-      }
+      await this._sender.close();
```

Comment:
Do we also need to cleanup it's entry from connectionContext.senders?

Reply:
The `MessageSender.close()` calls a helper method called `_deleteFromCache()` that removes the receiver from the context cache

Reply:
I have now been over that code for other recreational purposes and understand. ;)